packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -361,6 +371,59 @@ ListButton.propTypes = {
   }).isRequired,
 };
 
+const LinkButton = () => {
+  const [linkNode, setLinkNode] = React.useState(null);
+  const editor = useSlate();
+
+  const isLinkActive = () => {
+    if (!editor.selection) return false;
+
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        at: Editor.unhangRange(editor, editor.selection),
+        match: (node) =>
+          !Editor.isEditor(node) && SlateElement.isElement(node) && node.type === 'link',
+      })
+    );
+
+    return Boolean(match);
+  };
+
+  const isActive = isLinkActive();
+
+  const openLinkPopover = () => {
+    // We insert an empty anchor, so we split the DOM to have a element we can use as reference for the popover
+    insertLink(editor, { url: '', text: Editor.string(editor, editor.selection) });
+
+    const [parentNode] = Editor.parent(editor, editor.selection);
+
+    setLinkNode(parentNode);
+  };
+
+  React.useEffect(() => {
+    if (linkNode) {
+      const domNode = ReactEditor.toDOMNode(editor, linkNode);
+      domNode.click();
+    }
+  }, [linkNode, editor]);
```

Comment:
Do we have any examples from the `slate` repo how to do this? This doesn't look like _anything_ i've ever seen in relation to using slate or how any other competitor uses slate to do this sort of thing either.

There might be an example in `react-slate` on a "floating" toolbar which would be the same thing in theory.